### ISSUE TITLE
Rule idea: dict get for ranking dict

### ISSUE THREAD

**Issue by entorb** [astral-sh/ruff#14592]
Do you think a rule for improving code of ranking-dict increment would be worth implementing?

this
```python
if "key" not in d:
    d["key"] = 1
else:
    d["key"] += 1
```
should be simplified by
```python
d["key"] = d.get("key", 1)
```

**Comment by entorb** [comment]
Ups, yes of cause you are right, as I messed the second part up. Sorry for the confusion. Here corrected:

this
```python
if "key" not in d:
    d["key"] = 1
else:
    d["key"] += 1
```
should be simplified by
```python
d["key"] = d.get("key", 0) + 1
```

**Comment by dylwil3** [comment]
Great, thank you! I think, as I said above, that I would personally find this lint confusing. But of course I'm happy to be overruled by the community 😄

**Comment by MichaReiser** [comment]
Thanks for the suggestion 

One acceptance criteria for our rules is that the rule should be useful for most Python programmers and that there's enough support in the community to consider it good practice: 

> Generally, if a lint is triggered, this should be useful to most Python (3+) programmers seeing it most of the time.

I don't think there's enough evidence for avoiding the `if`...`else` branching in the above example to be considered "simpler".
